packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,864 @@
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+class RegularWindow extends StatefulWidget {
+  /// Creates a regular window widget.
+  /// [controller] the controller for this window
+  /// [child] the content to render into this window
+  /// [key] the key for this widget
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  RegularWindow({super.key, required this.controller, required this.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+  }
+
+  /// Controller for this widget.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  final RegularWindowController controller;
+
+  /// The content rendered into this window.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  final Widget child;
+
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  @override
+  State<RegularWindow> createState() => _RegularWindowState();
+}
+
+class _RegularWindowState extends State<RegularWindow> {
+  @override
+  void dispose() {
+    super.dispose();
```

Comment:
Hypernit: generally we put `super.dispose();` as the last statement in the `dispose` method. This is likely fine.